src/map-app/app/view/sidebar/directory.ts

```diff
@@ -68,11 +68,11 @@ export class DirectorySidebarView extends BaseSidebarView {
       d3.selectAll("li.sea-directory-field").attr("hidden", null);
     else {
       const a = d3.selectAll("li.sea-directory-field");
-      a.attr("hidden", null); // Hide everything
+      a.attr("hidden", true); // Hide everything
       a.filter(function () {
         // Unhide elements which have matching text
         return d3.select(this).text().toLowerCase().includes(input);
-      }).attr("hidden", true);
+      }).attr("hidden", null);
```

Comment:
I'm not even sure if this feature is meant to exist, because it's hidden in the UI and hard to find it with the cursor (just below the Directory header).

But I made this change because it was broken and it was an easy fix. The old behaviour was in reverse:
`a.attr("hidden", null);` unhid everything and then the ones that matched were actually hidden by
```
a.filter(function () {
        // Unhide elements which have matching text
        return d3.select(this).text().toLowerCase().includes(input);
      }).attr("hidden", true);
```

Swapping the nulls and trues reversed the behaviour to match what the code comments say.

Reply:
I think it's a vestige.  Maybe to be removed, but I've been cautious about just removing stuff without understanding. (My understanding is still often incomplete.)

Re-reading, ignore points 3 - 5, I think it does invert correctly.

Reply:
Ok I'll keep it for now then